service/config_provider.go

```diff
@@ -92,8 +92,9 @@ type ConfigProviderSettings struct {
 	// MapConverters is a slice of config.MapConverterFunc.
 	MapConverters []config.MapConverterFunc
 
-	// The configunmarshaler.ConfigUnmarshaler to be used to unmarshal the config.Map into config.Config.
-	// It is required to not be nil, use configunmarshaler.NewDefault() by default.
+	// Deprecated: [v0.49.0] because providing custom ConfigUnmarshaler is not necessary since users can wrap/implement
+	// ConfigProvider if needed to change the resulted config. This functionality will be kept for at least 2 minor versions,
+	// and if nobody express a need for it will be removed.
```

Comment:
```suggestion
	// Deprecated: [v0.49.0] providing custom ConfigUnmarshaler is not necessary since users can wrap/implement
	// ConfigProvider if needed to change the resulted config. This functionality will be kept for at least 2 minor versions,
	// and if no one expresses a need for it will be removed.
```